tests/pruners_tests/test_successive_halving.py

```diff
@@ -108,6 +108,35 @@ def test_successive_halving_pruner_with_nan():
     assert pruner.prune(study=study, trial=study._storage.get_trial(trial._trial_id))
 
 
+def test_successive_halving_pruner_with_auto_min_resource():
+    # type: () -> None
+
+    pruner = optuna.pruners.SuccessiveHalvingPruner(min_resource='auto')
+    study = optuna.study.create_study(sampler=optuna.samplers.RandomSampler(), pruner=pruner)
+
+    assert pruner._min_resource is None
+
+    def objective(trial):
+        # type: (optuna.trial.Trial) -> float
+
+        for i in range(3):
+            trial.report(1.0 / (i + 1))
+            if trial.should_prune():
+                raise optuna.exceptions.TrialPruned()
+        return 1.0
+
+    study.optimize(objective, n_trials=3)
+    # FIXME (crcrpar)
+    # assert pruner._min_resource is not None and pruner._min_resource > 0
```

Comment:
currently (https://github.com/optuna/optuna/pull/812/commits/c292d58a78042ec1122d78dd00292fd7dd8b46e6), this fails